src/lib/orion_websocket/parser.cpp

```diff
@@ -84,4 +87,49 @@ void ws_parser_parse
   {
     head.gotHeaders = false;
   }
+
+  if (doc.HasMember("params") && doc["params"].IsObject())
+  {
+    rapidjson::Value::ConstMemberIterator it = doc["params"].MemberBegin();
+    while (it != doc["params"].MemberEnd())
+    {
+      uriParamPush(ciP, it->name.GetString(), it->value.GetString());
+      ++it;
+    }
+  }
+}
+
+const char *ws_parser_message
+(
+ const std::string&  msg,
+ const HttpHeaders&  head,
+ int                 statusCode
+)
+{
+  const char *tmpl = "{\"headers\": %s, \"message\": %s, \"status\": \"%d\"}";
+
+  rapidjson::StringBuffer buff;
+  rapidjson::Writer<rapidjson::StringBuffer> writer(buff);
+  writer.StartObject();
+  std::map<std::string, std::string *>::const_iterator it = head.headerMap.begin();
+  while (it != head.headerMap.end())
+  {
+    writer.Key(it->first.c_str());
+    writer.String(it->second->c_str());
+    ++it;
+  }
+  writer.EndObject();
+
+  if (statusCode < 100 || statusCode > 599) // Code is not a valid HTTP status code
+    statusCode = 0;
```

Comment:
If the code comes wrong for some unknown reason, then it's a bug in the broker. Right?
500 seems like a good code for that.

#fgalan: this code is set **only** if the code is invalid, otherwise it goes thru as is.